sklearn/utils/estimator_checks.py

```diff
@@ -402,21 +402,55 @@ def pairwise_estimator_convert_X(X, estimator, kernel=linear_kernel):
     return X
 
 
+def _generate_sparse_matrix(X_csr):
+    """
+        Generate matrices in multiple formats
+
+        For format belonging to CSR,CSC,COO
+        it generates pair belonging to int64 
+        and int32 indices type.
+
+        Parameters
+        ----------
+
+        X_csr: CSR Matrix 
+            Input matrix in CSR format
+
+        Returns
+        -------
+
+        out: iter(Matrices)
+            In format ['dok','lil','dia','bsr','csr','csc','coo','csr_64','csc_64','coo_64']
+    """
+
+    for sparse_format in ['dok', 'lil', 'dia', 'bsr', 'csr', 'csc', 'coo']:
+        yield X_csr.asformat(sparse_format)
+
+    for sparse_format in ['csr', 'csc']:
+        X = X_csr.asformat(sparse_format)
+        X.indices = X.indices.astype('int64')
+        X.indptr = X.indptr.astype('int64')
+        yield X
```

Comment:
Maybe you should yield a name for each format also, and report it in the error message to be extra helpful to implementors

Reply:
I think we should do this, especially because accept_large_sparse deafaults to False.

Reply:
Do you need help completing this?